rest_framework_nested/routers.py

```diff
@@ -69,19 +82,21 @@ def __init__(self, parent_router, parent_prefix, *args, **kwargs):
             # we set our trailing slash to just '/', leading to inconsistent behavior.
             self.trailing_slash = parent_router.trailing_slash
 
-        parent_registry = [registered for registered
-                           in self.parent_router.registry
-                           if registered[0] == self.parent_prefix]
+        parent_registry: list[tuple[str, type[ViewSetMixin], str]] = [
```

Comment:
This comes from upstream code. What about signaling just what we know to be true?
```suggestion
        parent_registry: list[tuple] = [
```

My concern is that upstream changes and this breaks our type checks.

Reply:
> My concern is that upstream changes and this breaks our type checks.

Well then it would break the code as well. And it would be good to know when this code needs to be updated.

The tuple gets unpacked a few lines below, you're already very much dependent on these types:

```python
parent_registry_item = parent_registry[0]
parent_prefix, parent_viewset, parent_basename = parent_registry_item
```

Reply:
Actually never mind, I can remove this whole hint and mypy will infer it.